--- src/main/org/apache/tools/ant/util/JavaEnvUtils.java.orig	2015-11-27 14:57:37.135098404 +0100
+++ src/main/org/apache/tools/ant/util/JavaEnvUtils.java	2015-11-27 15:03:33.291330987 +0100
@@ -90,6 +90,21 @@
     /** Number Version constant for Java 1.6 */
     public static final int VERSION_1_6 = 16;
 
+    /** Version constant for Java 1.7 */
+    public static final String JAVA_1_7 = "1.7";
+    /** Number Version constant for Java 1.7 */
+    public static final int VERSION_1_7 = 17;
+
+    /** Version constant for Java 1.8 */
+    public static final String JAVA_1_8 = "1.8";
+    /** Number Version constant for Java 1.8 */
+    public static final int VERSION_1_8 = 18;
+
+    /** Version constant for Java 1.9 */
+    public static final String JAVA_1_9 = "1.9";
+    /** Number Version constant for Java 1.9 */
+    public static final int VERSION_1_9 = 19;
+
     /** Whether this is the Kaffe VM */
     private static boolean kaffeDetected;
 
@@ -125,9 +140,18 @@
             Class.forName("java.net.Proxy");
             javaVersion = JAVA_1_5;
             javaVersionNumber++;
-            Class.forName("java.util.ServiceLoader");
+            Class.forName("java.net.CookieStore");
             javaVersion = JAVA_1_6;
             javaVersionNumber++;
+            Class.forName("java.nio.file.FileSystem");
+            javaVersion = JAVA_1_7;
+            javaVersionNumber++;
+            Class.forName("java.lang.reflect.Executable");
+            javaVersion = JAVA_1_8;
+            javaVersionNumber++;
+            Class.forName("javax.lang.model.SourceVersion").getDeclaredField("RELEASE_9");
+            javaVersion = JAVA_1_9;
+            javaVersionNumber++;
         } catch (Throwable t) {
             // swallow as we've hit the max class version that
             // we have
